bot/resources/tags/defaultdict.md

```diff
@@ -0,0 +1,20 @@
+**[`collections.defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict)**
+
+The Python `defaultdict` type behaves almost exactly like a regular Python dictionary, but if you try to access or modify a missing key, the `defaultdict` will automatically create the key and generate a default value for it. 
+While instantiating a `defaultdict`, we pass in a function that tells it how to create a default value for missing keys.
+
+```py
+>>> from collections import defaultdict
+>>> my_dict = defaultdict(int, {"foo": 1, "bar": 2})
```

Comment:
I don't think it's very popular to instantiate a defaultdict from another dict, so it may be more useful to just have `my_dict = defaultdict(int)`. Otherwise people might erroneously assume that passing an empty dict is necessary if they don't need any key-values to start, or something.